### PR TITLE
Contacts translation page to design system.

### PR DESCRIPTION
**Pull request by farahTW**
This PR transits contact translation page to design system.

**It does the following:**
- Prepend legacy to existing edit view and controller test files.
- Add new index view to select translation language.
- Add new edit view for adding/editing existing translations.
- Updated the tests and added some new controller tests

**Screen shots:**
**Contacts Index page:**
image
image

**New translation index page:**
image

**New edit page:**
image

**Translation confirm destroy page:**
image

**Error scenario:**
image

**Success scenario:**
image

**Trello:**
https://trello.com/c/TSFOLQKo/464-add-new-endpoints-for-contacts-translation

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -85,11 +85,12 @@ def external_redirect(path_prefix, target)
             resources :translations, controller: "corporate_information_pages_translations"
           end
           resources :contacts do
-            resources :translations, controller: "contact_translations", only: %i[create edit update destroy] do
+            resources :translations, controller: "contact_translations", only: %i[create edit update destroy ] do
```
Comment (davidgisbey) [inline config/routes.rb]: you can just add the index here 

```
 %i[create edit update destroy index] do
```

File: app/controllers/admin/contact_translations_controller.rb [hunk 2]
```diff
@@ -2,16 +2,17 @@ class Admin::ContactTranslationsController < Admin::BaseController
...
     render_design_system(:edit, :legacy_edit)
   end
+
 private
 
   def get_layout
-    design_system_actions = %w[index confirm_destroy]
-    design_system_actions += %w[edit] if preview_design_system?(next_release: false)
+    design_system_actions = %w[confirm_destroy]
+    design_system_actions += %w[edit index] if preview_design_system?(next_release: false)
```
Comment (davidgisbey) [inline app/controllers/admin/contact_translations_controller.rb:15]: i think this was right before wasn't it? The index view is only ever in the GOV.UK Design System?

File: app/views/admin/contact_translations/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        right_to_left: @translated_contact.translation_locale.rtl?,
+        right_to_left_help: false
+      } %>
+
+      <h2 class="app-view-translation__english-content govuk-heading-m govuk-!-margin-bottom-2">English
+        translation:</h2>
+      <p class="app-view-translation__english-content govuk-body"><%= @translated_contact.comments %></p>
+      <%= render "govuk_publishing_components/components/input", {
+        label: {
+          text: "Recipient (required)",
```
Comment (davidgisbey) [inline app/views/admin/contact_translations/edit.html.erb:44]: I think the only required fields on this are title and street address 
image

File: app/controllers/admin/contact_translations_controller.rb (hunk 2 above)
Comment (farahTW) [reply app/controllers/admin/contact_translations_controller.rb:15]: ya but there was some issue while running the controller test so I changed it

**Comment by nnagewad** [comment]
Looks good from Design side

File: app/views/admin/contact_translations/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          {
+            text: language,
+            value: value,
+          }
+        end
+      } %>
+
+      <%= render "govuk_publishing_components/components/button", {
+        text: "Next"
+      } %>
```
Comment (davidtrussler) [inline app/views/admin/contact_translations/index.html.erb:25]: We will need a "cancel" link here.

**Comment by davidtrussler** [comment]
@farahTW Looks good except one small comment about a missing "Cancel" link. 😄

**Comment by nnagewad** [comment]
Actually. This was brought to my attention from @davidtrussler. You are missing a cancel link when choosing a language.

**Pull request merged**